packages/peregrine/src/hooks/useCarousel.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useCallback } from 'react';
+
+export const useCarousel = carouselSize => {
```

Comment:
Though I left feedback in here about the current implementation I think it makes sense to rethink this approach.

Right now this hook just handles the state and callbacks around setting the active item index. It could probably internalize some of the specifics to carousels. For example we could move the sorting into the carousel and accept an array of images rather than just the size.

```js
const [state, api] = useCarousel(images)
const { activeItem } = state;
const { handleNext, handlePrevious, setActiveItemIndex } = api;

...
}
```